File: ProblemCollection/src/monthly_code_challenge_season3/FindNumberWhereRemainderIs1.py
def solution(n):
    for i in range(2, n // 2):
    # for i in range(2, int(n ** .5) + 1):
        if n % i == 1: return i
    return n - 1
    # return next(i for i in range(2, n) if n % i == 1)

File: ProblemCollection/src/monthly_code_challenge_season3/FindNumberWhereRemainderIs1.java
package monthly_code_challenge_season3;

public class FindNumberWhereRemainderIs1 {
//	나머지가 1이 되는 수 찾기
	public int solution(int n) {
		for (int i = 2;; i++) {
			if (n % i == 1)
				return i;
		}
//        for(int i = 2; i < n/2; i++){
//            if(n%i == 1){
//                return i;
//            }
//        }
//        return n-1;
	}
}